primefaces-selenium/primefaces-selenium-components/src/main/java/org/primefaces/selenium/component/SelectManyMenu.java

```diff
@@ -103,9 +106,17 @@ public boolean isSelected(String label) {
     }
 
     public List<String> getLabels() {
-        return getInput().findElements(By.tagName("option")).stream()
+        if (getWidgetConfiguration().has("filter") &&  getWidgetConfiguration().getBoolean("filter")) {
```

Comment:
For efficiency maybe only get the widgetConfiguration once and store in a variable?